apps/backend/src/mutations/QuestionaryMutations.ts

```diff
@@ -53,9 +66,107 @@ export default class QuestionaryMutations {
     const questionIds: string[] = stepQuestions.map(
       (question) => question.question.id
     );
+
+    const genericTemplateQuestions = stepQuestions.filter(
+      (step) => step.question.dataType == DataType.GENERIC_TEMPLATE
+    );
+
+    if (genericTemplateQuestions) {
+      // Confirm dependency with dependency condition
+      const unsatisfiedDependencyQues = genericTemplateQuestions.filter(
+        (genericTemplateQues) => {
+          const notSatisfiedQuestions = genericTemplateQues.dependencies.filter(
+            (dependency) => {
+              const answer = answers.find(
+                (answer) => answer.questionId === dependency.dependencyId
+              );
+
+              if (answer) {
+                const { value } = JSON.parse(answer.value);
+                if (value[0] !== dependency.condition.params) {
+                  return true;
+                }
+              }
+            }
+          );
+
+          if (notSatisfiedQuestions.length != 0) {
+            return true;
+          }
+        }
+      );
+
+      unsatisfiedDependencyQues.forEach(async (subTemplateQues) => {
+        const config = subTemplateQues.config as SubTemplateConfig;
+        await this.deleteSubTemplatesAnswers(

```

Comment:
This will not be awaited, because `forEach` does not await the function body statements.
Maybe use `Promise.all` with map or a `for ... of`
This might now show as a problem if you do not expect immediate update latter in the code.